src/objects/set.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2017-present PyO3 Project and Contributors
+//
+use std::{hash, collections};
+use ffi;
+use python::{Python, PythonObject};
+use conversion::ToPyObject;
+use objects::{PyObject, PyIterator};
+use err::{self, PyResult, PyErr};
+
+
+/// Represents a Python `set`
+pub struct PySet(PyObject);
+/// Represents a  Python `frozenset`
+pub struct PyFrozenSet(PyObject);
+
+pyobject_newtype!(PySet, PySet_Check, PySet_Type);
+pyobject_newtype!(PyFrozenSet, PyFrozenSet_Check, PyFrozenSet_Type);
+
+impl PySet {
+    /// Creates a new set.
+    ///
+    /// May panic when running out of memory.
+    pub fn new(py: Python, elements: &[PyObject]) -> PySet {
```

Comment:
But other things like `PyList` still uses `pub fn new(py: Python, elements: &[PyObject]) -> PyList`, I guess we should do this in a separate PR to do them all.